extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/visitors/PanacheEntityClassAccessorGenerationVisitor.java

```diff
@@ -239,4 +243,7 @@ private boolean isAnnotatedWithJsonIgnore(EntityField entityField) {
         return false;
     }
 
+    private boolean hasGetterForField(EntityField entityField) {
+        return entityInfo.firstMethod(entityField.getGetterName()) != null;
```

Comment:
This is a bit optimistic because it doesn't check that it's a getter method (zero parameter, returns the same type).

Reply:
Ah, indeed I should use `method` instead.

As to checking for the return type... I'll try to implement a check, but I'll have to retrieve a "type descriptor" from the return type (to compare it with the field's type descriptor), and I'm not sure how easy that is.

Relatedly, assuming there's a `getMyProperty()` method, but its type is different, we will generate a new getter returning the field type. See https://github.com/quarkusio/quarkus/blob/51cb9667c303d62b5c7ffb0c40e47f2d57213f42/extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/visitors/PanacheEntityClassAccessorGenerationVisitor.java#L128-L130. So we'll have two getters with different return types. I'd say we are screwed in that case anyway? :thinking:

Reply:
Hah. Good point. Not sure what to do in this case. It never happened. Let's pretend it won't happen until it does ;)

Just check the signature, I guess?